internal/command/local/connect.go

```diff
@@ -264,6 +273,40 @@ doneWaiting:
 	return fmt.Errorf("connect failed and is no longer running")
 }
 
+func checkVersionSkew(localConfig *config.LocalConnect, ciConfig *config.ConnectInvocationConfig) error {
+	// get the cluster from API
+	params := clusters.NewGetClusterParams().
+		WithOrgName(localConfig.Org).WithClusterName(ciConfig.ConnectionConfig.Cluster)
+	clusterInfo, err := localConfig.Client.Cluster.GetCluster(params, nil)
+	if err != nil {
+		return fmt.Errorf("error reading cluster, %w", err)
+	}
+	if clusterInfo.Payload.Operator == nil {
+		return fmt.Errorf("cluster=%s has never connected with Signadot control-plane",
+			ciConfig.ConnectionConfig.Cluster)
+	}
+
+	// parse the operator version
+	operatorVer, err := semver.NewVersion(strings.Split(clusterInfo.Payload.Operator.Version, " ")[0])
+	if err != nil {
+		return fmt.Errorf("error parsing cluster operator version, %w", err)
+	}
+
+	if ciConfig.ConnectionConfig.Type == connectcfg.ControlPlaneProxyLinkType {
+		// ControlPlaneProxy requires operator > 0.15.0, in the constraint below
+		// we use "> 0.15.1-0" instead of ">= 0.16.0" to catch our pre-release
+		// versions
+		cppConstraint, err := semver.NewConstraint("> 0.15.1-0")
+		if err != nil {
+			return err // this shouldn't happen
+		}
+		if !cppConstraint.Check(operatorVer) {
+			return fmt.Errorf("The connection type ControlPlaneProxy requires operator > v0.15.0")
```

Comment:
I would prefer the message have `>=` instead of `>`.   We can also use server.Parse(v)+ .{Major,Minor,Patch}() as a cleaner way of checking in way consistent with our perversions.

Reply:
> I would prefer the message have >= instead of >

Sure, I can change it to ">= v0.16.0"

> We can also use server.Parse(v)+ .{Major,Minor,Patch}() as a cleaner way of checking in way consistent with our perversions.

I didn't get this point, can you explain more?

Reply:
Given a semver `v`, we can check if it has a [`Prelease`](https://pkg.go.dev/github.com/Masterminds/semver/v3#Version.Prerelease) and if so, let it pass, otherwise, otherwise default to the major, minor, and patch components.  

Or, since we know the version history, we can strip the prerelease and then check `>` (leaving the error message with >=).

Personally, I find these alternatives more explicit and readable. 

(autocorrection changed `semver` to `server` which might have caused some confusion too)

Reply:
@scott-cotton, is it better now? https://github.com/signadot/cli/pull/112/commits/e316dfef93fa47eaa0d744276ee1d1a45302242e